lib/resty/mlcache.lua

```diff
@@ -85,12 +96,12 @@ local marshallers = {
     end,
 
     [4] = function(t)      -- table
-        local json, err = cjson.encode(t)
-        if not json then
+        local str, err = encode(t)
```

Comment:
These new APIs can throw errors:

> This function may throw an error when fed with malformed or incomplete encoded data. The stand-alone function throws when there's left-over data after decoding a single top-level object. The buffer method leaves any left-over data in the buffer. 

I'd prefer we update the code here to use a protected call to encode (and decode) - we can require `cjson` instead of `cjson.safe` on the fallback encoder to guarantee the same behavior.

Reply:
OK, I will try it.